src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -97,8 +98,23 @@ public class RequestExecutor : IDisposable
 
         public event EventHandler<(long RaftCommandIndex, ClientConfiguration Configuration)> ClientConfigurationChanged;
 
+        public event Action<string> SucceededRequest;
+        public event Action<string, HttpRequestException> FailedRequest;
+
+        protected void OnSucceededRequest(string url)
+        {
+            SucceededRequest?.Invoke(url);
+        }
+
+        protected void OnFailedRequest(string url, HttpRequestException e)
```

Comment:
Why do we have `nodeSelector?.OnFailedRequest(nodeIndex);` vs `OnFailedRequest(url,e);`